src/modals/JoinSponsorshipModal.tsx

```diff
@@ -274,7 +284,40 @@ function JoinSponsorshipModal({
                     </li>
                     <li>
                         <Prop>Operator</Prop>
-                        <PropValue>{operatorId}</PropValue>
+                        <PropValue>
+                            {operatorChoices?.length === 1 ? (
+                                <>
+                                    {metadata.name != null
+                                        ? `${metadata.name} (${truncate(operatorId)})`
+                                        : operatorId}
+                                </>
+                            ) : (
+                                <>
+                                    <SelectField2
```

Comment:
If address controls multiple Operators we'll show a dropdown for operator selection